inc/functions.php

```diff
@@ -5033,6 +5017,43 @@ function build_theme_select($name, $selected="", $tid=0, $depth="", $usergroup_o
 }
 
 /**
+ * Get the theme data of a theme id.
+ *
+ * @param int The theme id of the theme.
+ * @return array The theme data
+ */
+function get_theme($tid)
+{
+	global $tcache, $db;
+
+	if(!is_array($tcache))
+	{
+		$query = $db->simple_select('themes', 'tid, name, pid, allowedgroups', "pid!='0'", array('order_by' => 'pid, name'));
```

Comment:
Just noticed, any reason for `array('order_by' => 'pid, name')` here? Why would there be a need to sort the results by any of these fields? The `while` loop will go through all of them anyways and do the same thing. So it only slows down the query.